app/commands/cleanup.rb

```diff
@@ -0,0 +1,64 @@
+require 'optparse'
+
+module Commands
+  class Cleanup
+    include Logging
+
+    attr_accessor :etcd_index_upto
+    attr_accessor :docker_networks
+    attr_accessor :pool
+    attr_accessor :addresses
+
+    def self.parse!
+      command = new
+
+      OptionParser.new do |opts|
+        opts.banner = "Usage: kontena-ipam-cleanup [--docker-networks] [--pool POOL IPADDR...]"
+        opts.on("--etcd-index-upto=INDEX", Integer, "Only cleanup nodes created <= given etcd index") do |index|
```

Comment:
Some sane default would be really nice for this, e.g. `current_index - 10`. That way agent, and any other possible user, would not really have to understand ETCD indexes at all.